apps/champions/lib/champions/users.ex

```diff
@@ -118,4 +145,44 @@ defmodule Champions.Users do
   def calculate_experience_to_next_level(level) do
     Math.pow(100, 1 + level / 10) |> ceil()
   end
+
+  @doc """
+  Get a user's available AFK rewards, according to their AFK reward rates and the time since their last claim.
+  If more than 12 hours have passed since the last claim, the user will have accumulated the maximum amount of rewards.
+  """
+  def get_afk_rewards(user_id) do
+    user = Users.get_user(user_id)
+
+    user.afk_reward_rates
+    |> Enum.map(fn reward_rate ->
+      currency = Currencies.get_currency(reward_rate.currency_id)
+      amount = calculate_afk_rewards(user, reward_rate)
+      %{currency: currency, amount: amount}
+    end)
+  end
+
+  defp calculate_afk_rewards(user, afk_reward_rate) do
```

Comment:
Here we have an issue where if we accumulate a bunch of hours with a low rate (say, 10g/m), and before claiming we beat a bunch of levels getting our rate to something like 90g/m, we'll get the totality of hours on the new rate, resulting in a broken system. Check out how we've implemented this in the frontend! I can give a hand with the logic as well